haskell/adventure.hs

```diff
@@ -0,0 +1,426 @@
+    ("ChataDrwali", Location [] [("wyjdź", "PolanaKołoChatyDrwali")]),
+    ("Puszcza", Location [] [("polana", "PolanaKołoChatyDrwali"), ("pieczara", "Pieczara")]),
+    ("Pieczara", Location [] [("wyjdź", "Puszcza")])
+  ]
+
+-- game state
+
+data GameState = GameState
+  { location :: String,
+    inventory :: Set Item,
+    itemLocations :: Map LocationName (Set Item),
+    facts :: Set String,
+    hp :: Int,
+    enemyHp :: Int,
+    timeOfDay :: TimeOfDay,
+    dayTurnCounter :: Int,
+    attackStrength :: Int
+  }
+
+initialState :: GameState
+initialState =
+  GameState
+    { location = "ŚwiętyDąb",
+      inventory = Set.empty,
+      itemLocations = [
+        ("ŚwiętyDąb", ["Ciało", "Totem"]),
+        ("WieśGrobla", ["Studnia", "TablicaOgłoszeń"]),
+        ("DomSołtysa", []),
+        ("Karczma", []),
+        ("PolanaKołoChatyDrwali", ["CidaryjskaPrzeszywanica", "Trup"]),
+        ("ChataDrwali", ["Topór", "Czaszka", "Mieszek"]),
+        ("Puszcza", ["Statua"])
+      ],
+      facts = Set.empty,
+      hp = 100,
+      enemyHp = 100,
+      timeOfDay = Day,
+      dayTurnCounter = 0,
+      attackStrength = 10
+    }
+
+-- game actions
+
+describeItem :: Item -> String
+describeItem "Totem" = "Analiza Gerwanta z Riviery: Totem został zbudowany ze szczątek i poroża jelenia. To wygląda na ostrzeżenie."
+describeItem "Ciało" = "Analiza Wiedźmina: Rany od szponów i kłów."
+
+describeItem "Studnia" = " Rivierijczyk zauważa *gnomskiGwyhyr* na dnie studni."
+describeItem "GnomskiGwyhyr" = "Gwyhyr wykuty przez gnomy. Jest sprawnie naostrzony. Na klindze ma wyryte runy."
+describeItem "TablicaOgłoszeń" = "Nie ma takiego przedmiotu."
+
+describeItem "CidaryjskaPrzeszywanica" = "Przeszywanica popularna pośród piechurów w Cidaris"
+describeItem "Trup" = "Analiza Gerwanta z Riviery: \"Poturbowany i mocno poobijany. Krew nie została wyssana, ale był pogryziony przez wilki\""
+
+describeItem "Topór" = "Analiza Gerwanta z Riviery: \"W rysach topora zastała się zaschnięta krew zwierzęcia\""
+describeItem "Czaszka" = "Gerwanta z Riviery: \"Ta czaszka należy do młodego niedźwiedzia\""
+describeItem "Mieszek" = "Mieszek pełen złota. W środku znajduje się 10 monet. Wystarczy na piwo."
+
+describeItem "Statua" = " Analiza Gerwanta z Riviery: \"Medalion Drży. To magiczny totem. Podobny totem widziałem pod Świętym dębem\". Wiedźmin postanawia przeczytać bestiariusz. W wiedźmińskim almanachu Gerwant znajduje informację, że magiczne totemy to znak rozpoznawczny terytorium potwora zwanego jako *Leszy*."
+describeItem _ = "Nie ma takiego przedmiotu."
+
+describeLocation :: String -> String
+describeLocation "ŚwiętyDąb" = "Gerwant z Riviery, szkoły nosacza, podróżuje szlakami Królestw Północy już wiele dni.\
+\Towarzyszy mu jedynie deszcz i jego wierny koń Piwonia.\
+\Pewnego popołudnia dociera do Świętego Dębu.\
+\Niestety te sakramentalne miejsce zostało zbrukane ludzką krwią."
```

Comment:
This is a game introduction, lets change it to ` "Gerwant dociera do Świętego Dębu.\
\Niestety te sakramentalne miejsce zostało zbrukane ludzką krwią." `